apps/web/src/pages/integrations/CreateProviderPage.tsx

```diff
@@ -0,0 +1,38 @@
+import { useState } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+import { useEffectOnce } from '../../hooks';
+import { IProviderConfig, providers } from '@novu/shared';
+import { CreateProviderInstanceSidebar } from './components/multi-provider/CreateProviderInstanceSidebar';
+import { SideBarWrapper } from './components/multi-provider/SelectProviderSidebar';
+
+export function CreateProviderPage() {
+  const [selectedProvider, setSelectedProvider] = useState<IProviderConfig | null>(null);
+  const { channel, providerId } = useParams();
+  const navigate = useNavigate();
+
+  useEffectOnce(() => {
+    const foundProvider = providers.find((provider) => provider.channel === channel && provider.id === providerId);
+
+    if (foundProvider) {
+      setSelectedProvider(foundProvider);
+    }
+  }, channel !== undefined && providerId !== undefined);
+
+  if (selectedProvider === null) {
+    return null;
+  }
```

Comment:
the selected provider doesn't change, so you don't need a state for that... what you can do instead is:
```
const foundProvider = useMemo(() => providers.find((provider) => provider.channel === channel && provider.id === providerId), [channel, providerId]);
```